master/internal/db/postgres_checkpoints.go

```diff
@@ -111,3 +114,27 @@ func (db *PgDB) GroupCheckpointUUIDsByExperimentID(checkpoints []uuid.UUID) (
 
 	return groupeIDcUUIDS, nil
 }
+
+// ExperimentConfigForCheckpoint looks up the experiment config of the checkpoint by the
+// checkpoint UUID, and it returns nil if the checkpoint does not exist.
+func (db *PgDB) ExperimentConfigForCheckpoint(id uuid.UUID) (*expconf.ExperimentConfig, error) {
+	checkpoint, err := db.CheckpointByUUID(id)
+	if err != nil {
+		return nil, err
+	}
+	if checkpoint == nil {
+		return nil, err
+	}
+
+	bytes, err := json.Marshal(checkpoint.CheckpointTrainingMetadata.ExperimentConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	var config expconf.ExperimentConfig
+	err = json.Unmarshal(bytes, &config)
+	if err != nil {
+	}
```

Comment:
Any time you are reading a legacy experiment config (a config which was valid when the experiment ran, but the experiment is no longer running and there's no telling how old the config is), you must unmarshal it as an `expconf.LegacyConfig` instead of an `expconf.ExperimentConfig`.  `ExperimentConfig` is really just for experiments which are currently running on the cluster, and after that we have to be able to handle experiment configs which were valid but no longer are.  Check out `expconf.ParseLegacyConfigJSON` which does this for you.

Separately, since I totally forgot that Checkpoint objects have experiment config information already, I'd recommend just using `db.CheckpointByUUID`, parsing the LegacyConfig in `m.getCheckpoint()`, and not adding this function to the db package at all.